### ISSUE TITLE
Matplotlib image does not apepar on margin in pdf

### ISSUE THREAD

**Issue by finkelshtein** [quarto-dev/quarto-cli#6863]
Hi,

I'd like to use Tufte style with Matplotlib images for both HTML and PDF. The following code produce expected HTML, however, in PDF, the image is placed in the main text, not at the margin.

````
---
title: "matplotlib demo"
format:
  html: 
      self-contained: true
      grid: 
          margin-width: 350px
  pdf: default
jupyter: python3
---

Demo

```{python}
#| echo: false
#| column: margin

import matplotlib.pyplot as plt

plot = plt.figure()
plot.set_figwidth(2)
plot.set_figheight(3)

x = [1, 2]
y = [2, 3]

plt.plot(x,y)

```
````

Looking at the Latex-file, I can see that \marginnote is just absent. When I try to enclose the code into another {.column-margin} div:

````
::: {.column-margin}
```{python}

code

```
::: 

````
it still works in HTML, but it makes error in PDF, as it produces Latex `figure` environment inside \marginnote.

1. Is it a bug, or I missed an option?
2. Is it possible to get an output of Matplotlib without placing it into `figure`?
3. Otherwise, I will probably need to export image from the Python code and import it inside `{.column-margin}`.

Many thanks!

**Comment by jjallaire** [comment]
cc @dragonstyle

**Comment by dragonstyle** [comment]
This appears to be related to some recent work in Quarto related to handling of figures and cross references. Thanks for letting us know we'll get it fixed asap! (worked in `1.4.358`).

**Comment by cscheid** [comment]
The bug is that `columns-preprocess.lua` is not correctly forwarding the classes into the custom AST nodes.

**Comment by cscheid** [comment]
Thank you for the report and reproduction! PR with fix is up on #6865, will be merged once tests pass.

**Comment by cscheid** [comment]
Thank you so much for the report again. We fixed the bug on `main` and will have a new prerelease binary shortly.
